src/gluonnlp/adapters.py

```diff
@@ -0,0 +1,247 @@
+            ('dropout', dropout),
+            ('normalization', normalization),
+            ('layer_norm_eps', layer_norm_eps),
+            ('pre_norm', pre_norm),
+            ('dtype', self._dtype)
+        ])
+        self.dropout_layer = nn.Dropout(dropout)
+        self.activation_dropout_layer = nn.Dropout(activation_dropout)
+        self.ffn_1 = nn.Dense(units=hidden_size,
+                              in_units=units,
+                              flatten=False,
+                              use_bias=use_bias,
+                              weight_initializer=weight_initializer,
+                              bias_initializer=bias_initializer,
+                              dtype=dtype)
+        if use_gated_activation:
+            self.gated_ffn_1 = nn.Dense(units=hidden_size,
+                                        in_units=units,
+                                        flatten=False,
+                                        use_bias=use_bias,
+                                        weight_initializer=weight_initializer,
+                                        bias_initializer=bias_initializer,
+                                        dtype=dtype)
+        self.activation = get_activation(activation)
+        self.ffn_2 = nn.Dense(units=units,
+                              in_units=hidden_size,
+                              flatten=False,
+                              use_bias=use_bias,
+                              weight_initializer=weight_initializer,
+                              bias_initializer=bias_initializer,
+                              dtype=dtype)
+        # TODO(sxjscience) We may need to set the dtype flag in LayerNorm, need to double check
+        self.layer_norm = get_norm_layer(in_channels=units,
+                                         normalization=normalization,
+                                         epsilon=layer_norm_eps,
+                                         **kwargs)
+        if self._use_adapter and 'location_1' in self._adapter_config:
+            self.adapter_layer_ffn = AdapterModule(in_units=units, adapter_config=adapter_config['location_1'])
+
+    def forward(self, data):
+        """
+
+        Parameters
+        ----------
+        F
+        data :
+            Shape (B, seq_length, C_in)
+
+        Returns
+        -------
+        out :
+            Shape (B, seq_length, C_out)
+        """
+        residual = data
+        if self._pre_norm:
+            data = self.layer_norm(data)
+        if self._use_gated_activation:
+            gated_out = self.activation(self.gated_ffn_1(data))
+            out = gated_out * self.ffn_1(data)
+        else:
+            out = self.activation(self.ffn_1(data))
+        out = self.activation_dropout_layer(out)
+        out = self.ffn_2(out)
+        out = self.dropout_layer(out)
+        if self._use_adapter and 'location_1' in self._adapter_config:
+            out = self.adapter_layer_ffn(out, residual)

```

Comment:
Based on your implementation of `BasicAdapter`, you'd need to call this layer as `self.adapter_layer_ffn(out, out)`.